yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -1021,20 +1022,23 @@ func fetchUnsupportedQueryConstructs() ([]utils.UnsupportedQueryConstruct, error
 	for i := 0; i < len(executedQueries); i++ {
 		query := executedQueries[i]
 		log.Debugf("fetching unsupported query constructs for query - [%s]", query)
-		queryParser := queryparser.New(query)
-		err := queryParser.Parse()
-		if err != nil {
-			log.Errorf("failed to parse query - [%s]: %v", query, err)
-		}
 
-		unsupportedConstructs, err := queryParser.GetUnsupportedQueryConstructs()
+		issues, err := parserIssueDetector.GetIssues(query)
 		if err != nil {
-			log.Errorf("failed while trying to fetch unsupported constructs from parse tree of query - [%s]: %s",
-				query, err.Error())
+			log.Errorf("failed while trying to fetch query issues in query - [%s]: %v",
+				query, err)
 		}
-		if unsupportedConstructs != nil {
-			result = append(result, unsupportedConstructs...)
+
+		for _, issue := range issues {
+			uqc := utils.UnsupportedQueryConstruct{
+				Query:             issue.SqlStatement,
+				ConstructType:     issue.Type,
+				ConstructTypeName: issue.TypeName,
```

Comment:
why do we need to have these two `ConstructType`, and `ConstructTypeName` separate fields?

Reply:
Thought process was:
`type` fields should be like enums (constants, no spaces, underscores). Currently `Advisory Locks` was a value being passed to `type` field, which seems wrong. 

I guess we need two things:
- `type` - a constant /enum determining which type of issue it is
- `typeName` - display name/title of the issue. 

So, I put both of these in the main `Issue` struct. And here in `UnsupportedQueryConstruct`, I figured it'd be good to put both (we use the `typeName` when displaying in html` ), but it's probably not required for json so i omitted it from that. I kept the `type` field also because it will be useful in json and also when we send to yugabyted. (the `typeName` is also probably useful for yugabyted)

Does that make sense?

Reply:
Yes, Agreed we should have both `Type` and `TypeName` in `Issue` but for the `Unsupported Query Construct` I am not sure if we should put both as it's not required as we can keep either of them, so as`Advisory Locks` is making more sense for `ConstructTypeName` field, so we can rename the `ConstructType` to that and keep only that which can be used by both json and html.

Reply:
To be consistent I think I agree it makes sense for us right now to just have the `ConstructTypeName` field in `UnsupportedQueryConstruct` . 

But this is something we should think about long term. IMO, in a json report, we should have `Type` for sure, and `TypeName` is optional.